src/test/java/demoqa/ElementsWebPageTest.java

```diff
@@ -0,0 +1,159 @@
+        clickMe.click();
+
+        WebElement textMessage = findElement("//p[@id='dynamicClickMessage']");
+
+        SoftAssert soft = new SoftAssert();
+        soft.assertEquals(textMessage.getText(),"You have done a dynamic click");
+        System.out.println(textMessage.getText());
+
+        soft.assertAll();
+    }
+
+    @Test
+    public void RegistrationFormFillingTest() throws Throwable {
+        this.wd().get(new URI(this.wd().getCurrentUrl()).resolve("/webtables").toString());
+
+        String xpathExpression = "//button[@id='addNewRecordButton']";
+        WebElement addButton = findElement(xpathExpression);
+        addButton.click();
+
+        WebElement registrationForm = findElement("//form[@id='userForm']");
+
+        WebElement firstName        = findElement("//input[@id='firstName']");
+        WebElement lastName         = findElement("//input[@id='lastName']");
+        WebElement email            = findElement("//input[@id='userEmail']");
+        WebElement age              = findElement("//input[@id='age']");
+        WebElement salary           = findElement("//input[@id='salary']");
+        WebElement department       = findElement("//input[@id='department']");
+        WebElement submitButton     = findElement("//button[@id='submit']");
+
+        if (registrationForm.isDisplayed()) {
+            firstName.clear();
+            firstName.sendKeys("Mark");
+            lastName.sendKeys("Thompson");
+            email.sendKeys("markThompson@test.ua");
+            age.sendKeys("35");
+            salary.sendKeys("35000");
+            department.sendKeys("Financial");
+
+            submitButton.click();
+        }
+
+        WebElement firstNameInTable     = findElement("//div[contains(@class, 'ReactTable')]//div[@role='row']//div[contains(text(), 'Mark')]");
+        WebElement lastNameInTable      = findElement("//div[contains(@class, 'ReactTable')]//div[@role='row']//div[contains(text(), 'Thompson')]");
+        WebElement emailInTable         = findElement("//div[contains(@class, 'ReactTable')]//div[@role='row']//div[contains(text(), 'markThompson@test.ua')]");
+        WebElement ageInTable           = findElement("//div[contains(@class, 'ReactTable')]//div[@role='row']//div[contains(text(), '35')]");
+        WebElement salaryInTable        = findElement("//div[contains(@class, 'ReactTable')]//div[@role='row']//div[contains(text(), '35000')]");
+        WebElement departmentInTable    = findElement("//div[contains(@class, 'ReactTable')]//div[@role='row']//div[contains(text(), 'Financial')]");
+        WebElement editButton           = findElement("//span[@id='edit-record-4']");
+
+        System.out.println("----------------------Verification of added values in the table----------------------");
+
+        SoftAssert soft = new SoftAssert();
+        soft.assertEquals(firstNameInTable.getText(),"Mark", "Incorrect first name is in table");
```

Comment:
Actually you are testing xpath here instead of table. You are using xpath `//div[contains(@class, 'ReactTable')]//div[@role='row']//div[contains(text(), 'Mark')]` so the text almost always will be `Mark`

Reply:
In this case even better to find row number by email and after that get cols by number and check that value is as expected, like that:
```
List<WebElement> emails = this.wd().findElements(By.xpath("//div[@role='row']/div[text()][4]"));
int addedRowNumber = -1;
for(int i = 0; i < email.size(); i++ ) 
   if (email.get(i).getText().equals("markThompson@test.ua")) {
      addedRowNumber  = i + 1;
      break;
   }
Assert.assertNotEquals(addedRowNumber , -1, "Record is not found by email");

...
WebElement firstNameInTable = findElement("(//div[@role='rowgroup'])[" + addedRowNumber + "]//div[@role='gridcell'][1]");
...
```